lib/PerceptionManager.tsx

```diff
@@ -59,47 +64,82 @@ export const perceptionManager: IPerceptionManager = (() => {
         });
     }
 
+    const buildSceneTurtleData = (things: Thing[]): Promise<string> => {
+        const scene: SolidDataset = createSolidDataset();
+
+        for(let thing of things) {
+            setThing(scene, thing);
+        }
+
+        return triplesToTurtle(Array.from(scene));
+    }
+
+    /**
+     * builds a list of ITerminalMessage objects by getting the primary content data from parameterised things
+     */
+    const getPrimaryContent = (things: Thing[]) : ITerminalMessage[] => {
+        let messages: ITerminalMessage[] = [];
+        
+        for(let thing of things) {
+            const uri = asUrl(thing as ThingPersisted);
+            const imageUrl = getUrl(thing, MUD.primaryImageContent);
+
+            if(imageUrl && !recentUris.includes(uri)) {
+                messages.push(getITerminalMessage(<img src={imageUrl}></img>));
+            }
+    
+            const msg = (getThingName(thing) + ". " || "") + (getStringNoLocale(thing, MUD.primaryTextContent) || "");
```

Comment:
What about:
```
const isPresent = value => Boolean(value)
const msg = [getThingName(thing), getStringNoLocale(thing, MUD.primaryTextContent)].filter(isPresent).join(". ")
```

That :point_up: wouldn't behave exactly the same, if getStringNoLocale was falsy but getThingName was:
Current Output: `thingname. `
Output with this change: `thingname`

But I think that's what you want?

Reply:
Nice style :)